### DISCUSSION TITLE
Measuring production rates of metabolites using FBA

### DISCUSSION THREAD

**Discussion by manas-kohli** [SysBioChalmers/Human-GEM#577] [Q&A]
Hey there!

Recently, I've been examining some papers on the application of FBA for cancer specific models. I recently came across this paper: https://www.nature.com/articles/s41467-021-22989-1 that looks at applying FBA in a machine learning context to discover cancer metabolic biomarkers. The key thing they do in their paper is generate their own metabolomics data from FBA by creating artificial sinks in the Recon3D network and then maximizing the objective function through these sinks. This approach sounded a bit interesting but raised a few questions for me:

1) Is estimating metabolite production in this way actually physiologically meaningful? The way I understand things on a naive level is that for many metabolites, you have steady state with flux being equal in and out. But if you add an artificial sink to your system and divert flux towards that, you disrupt equilibrium and in turn affect the whole system pretty drastically when it establishes itself at a new steady state. So should that not affect the entire FBA results and then any associations you see are just artefacts of this.

2) If the answer to 1) is that it won't affect the corresponding flux values, what's the way to actually add these sink reactions? I hunted the github repo to see where they do this but really couldn't find anything. If I were to take a Human1-GEM for instance, what would be the commands to add these artificial sink reactions and get estimates of metabolite levels in this way. 

Currently, I could just solve for flux by changing the objective function to sinks already present in the system but the whole concept of artifical sinks raised my curiosity to see whether this approach is applicable or whether the results are slightly spurious.

**Comment by haowang-bioinfo** [comment]
Thank you for introducing this interesting paper.

The best candidates to answer the questions probably are the authors of the paper. Very welcome to post here if you get the answers from them.

**Comment by mihai-sysbio** [comment]
Interesting question @manas-kohli.
To me, any attempt of pushing the model to the extremes (max/min) immediately introduces the question of reliability of the constraints (what would happen if the abstract upper bounds of the reactions are increased ten fold?). While the underlying method refers to incorporating _k<sub>cat</sub>_ values from BRENDA, what has been observed is that the coverage is low for the intended application. In [Fig S2](https://www.cell.com/cms/10.1016/j.cels.2020.12.001/attachment/d1188e43-2f80-4496-88fb-482a3d3e18c4/mmc1), only 0.19% are a perfect match. In my view, this highlights the unreliability of the constraints, and thus the unreliability of any model that is not finely tuned to match experimental data. Therefore, I would be very cautious in trusting absolute values. Instead, I would be more tempted to compare, e.g., metabolite levels when different objective functions are chosen.

**Comment by manas-kohli** [comment]
I managed to obtain a response from the lead author Dr. Kemp which is quite similar to the direction @mihai-sysbio suggested:

1) Estimating metabolite levels by iteratively setting the maximum production flux for that metabolite as an objective function is not physiologically meaningful, other than to demonstrate the capacity of the patient-specific system to generate that molecule. Our findings indicate that this capacity to reroute fluxes in order to generate a metabolite reflects to a certain degree the relative changes in concentrations of those metabolites across the samples. It is important to note that we did not interpret the fluxes that occur during this rerouting, only the relative changes in outputs in comparison the metabolomic data acquired.

2) The repo you want to use is associated with the Cell Systems paper (https://github.com/kemplab/FBA-pipeline) . In those instructions, Step 3.20 walks through the way to implement the metabolite screen per sample through the settting up of the template input file.

So if I understood correctly, the approach they applied and what I would need to do is to treat each exchange/demand reaction (https://metabolicatlas.org/explore/Human-GEM/gem-browser/subsystem/exchange_demand_reactions)  as an objective function and maximise flux through this in an enzyme constrained model as a proxy for ability at least to create the metabolite i.e. metabolite abundance. I will try this approach and hope that for many of these reactions, I don't run into infeasible loops and get values of 1000 etc. Do you think this sounds sensible in this context?
